sdk/rust/oak_config_serializer/src/main.rs

```diff
@@ -78,13 +87,44 @@ impl Default for InitialNodeConfig {
     }
 }
 
+fn check_sha256_sum(data: &Vec<u8>, sum: &str) -> bool {
+    let mut hasher = Sha256::new();
+    hasher.update(data);
+    let result_sum = hasher.finalize();
+    let extected_sum = hex::decode(sum).expect("Couldn't decode SHA256 sum");
+    extected_sum
+        .iter()
+        .zip(result_sum.as_slice())
+        .all(|(first, second)| first == second)
+}
+
 /// Load Wasm module from URL or a file, if URL is not specified.
-fn load_module(module: &Module) -> anyhow::Result<Vec<u8>> {
+async fn load_module(module: &Module) -> anyhow::Result<Vec<u8>> {
     match module {
-        Module::Url(_url) => unimplemented!(),
         Module::Path(path) => {
             fs::read(&path).with_context(|| format!("Couldn't read file {}", path))
         }
+        Module::External(external) => {
+            let url: Url = format!("{}/{}", &external.url, &external.sha256)
+                .parse()
+                .context("Couldn't parse URL")?;
+
+            debug!("Downloading module from: {}", url);
+            let response = reqwest::get(url)
```

Comment:
I guess at some point we need to have a cache of modules, since this will re-download them every time, perhaps add a TODO + issue?

Reply:
Added an issue https://github.com/project-oak/oak/issues/1240